addOns/retire/CHANGELOG.md

```diff
@@ -4,7 +4,9 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Changed
+- Performance improvements (Issue 6959).
```

Comment:
Did you check that this improves? (It should but would be good to confirm with a file with similar size.)

Reply:
Yes. I'll do some more live testing and find a message to include in unit tests I guess?

Reply:
I don't think it's necessary to add a real file, a dummy with random content should replicate too.

Reply:
I can't find a good one to save but here's my example:

```bash
Original

159972 [ZAP-PassiveScan-3] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 5 seconds to scan https://cdn.syndication.twimg.com/timeline/profile?callback=__twttr.callbacks.tl_i0_profile_HoCChamber_old&dnt=false&domain=www.ourcommons.ca&lang=en&screen_name=HoCChamber&suppress_response_codes=true&t=1838258&tz=GMT-0400&with_replies=false application/javascript;charset=utf-8 129242
160045 [ZAP-PassiveScan-3] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 5 seconds to scan https://cdn.syndication.twimg.com/timeline/profile?callback=__twttr.callbacks.tl_i0_profile_HoCChamber_old&dnt=false&domain=www.ourcommons.ca&lang=en&screen_name=HoCChamber&suppress_response_codes=true&t=1838258&tz=GMT-0400&with_replies=false application/javascript;charset=utf-8 129242
166568 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 15 seconds to scan https://platform.twitter.com/widgets/widget_iframe.d7fc2fc075c61f6fa34d79a0cbbf1e34.html?origin=https%3A%2F%2Fwww.ourcommons.ca text/html; charset=utf-8 327158
166694 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 9 seconds to scan https://www.parl.ca/legisinfo/en/overview text/html; charset=utf-8 67545
166726 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 9 seconds to scan https://www.parl.ca/legisinfo/en/overview text/html; charset=utf-8 67545
166756 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 16 seconds to scan https://platform.twitter.com/widgets/widget_iframe.d7fc2fc075c61f6fa34d79a0cbbf1e34.html?origin=https%3A%2F%2Fwww.ourcommons.ca text/html; charset=utf-8 327158
170400 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 28 seconds to scan https://www.ourcommons.ca/en text/html; charset=utf-8 289228
170610 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 28 seconds to scan https://www.ourcommons.ca/en text/html; charset=utf-8 289228
171939 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 5 seconds to scan https://www.parl.ca/CommonElementsV3/js/jquery-3.5.1.min.js application/javascript 89478
172002 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 5 seconds to scan https://www.parl.ca/CommonElementsV3/js/jquery-3.5.1.min.js application/javascript 89478
181226 [ZAP-PassiveScan-3] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 9 seconds to scan https://petitions.ourcommons.ca/en/Home/Index text/html; charset=utf-8 74747
181273 [ZAP-PassiveScan-3] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 9 seconds to scan https://petitions.ourcommons.ca/en/Home/Index text/html; charset=utf-8 74747
185080 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 8 seconds to scan https://petitions.ourcommons.ca/assets/site.css?v=2 text/css 268881
185083 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 8 seconds to scan https://petitions.ourcommons.ca/assets/site.css?v=2 text/css 268881
192056 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 14 seconds to scan https://www.gstatic.com/recaptcha/releases/M-QqaF9xk6BpjLH22uHZRhXt/recaptcha__en.js text/javascript 370206
192188 [ZAP-PassiveScan-4] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 14 seconds to scan https://www.gstatic.com/recaptcha/releases/M-QqaF9xk6BpjLH22uHZRhXt/recaptcha__en.js text/javascript 370206

195534 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 12 seconds to scan https://www.gstatic.com/recaptcha/releases/M-QqaF9xk6BpjLH22uHZRhXt/recaptcha__en.js text/javascript 370206
195663 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library took 12 seconds to scan https://www.gstatic.com/recaptcha/releases/M-QqaF9xk6BpjLH22uHZRhXt/recaptcha__en.js text/javascript 370206

251193 [AWT-EventQueue-0] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Installing new addon retire v0.13.0
251197 [AWT-EventQueue-0] INFO  org.zaproxy.zap.extension.pscan.ExtensionPassiveScan - loaded passive scan rule: Vulnerable JS Library (Retire.js)
251200 [AWT-EventQueue-0] INFO  org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate - Finished installing new addon retire v0.13.0

282437 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 6 seconds to scan https://petitions.ourcommons.ca/assets/site.css?v=2 text/css 268881
282439 [ZAP-PassiveScan-1] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library (Retire.js) took 6 seconds to scan https://petitions.ourcommons.ca/assets/site.css?v=2 text/css 268881
376124 [AWT-EventQueue-0] INFO  org.parosproxy.paros.control.Control - Discard Session
376169 [AWT-EventQueue-0] INFO  org.parosproxy.paros.control.Control - New Session
376170 [AWT-EventQueue-0] INFO  org.parosproxy.paros.control.Control - Create and Open Untitled Db
376191 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit start
376276 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit end
376332 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - Database closed
376507 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache open start
376632 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit start
376716 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit end
376716 [AWT-EventQueue-0] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache open end

New

398937 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Wappalyzer Scanner (Tech Detection) took 11 seconds to scan https://platform.twitter.com/widgets/widget_iframe.d7fc2fc075c61f6fa34d79a0cbbf1e34.html?origin=https%3A%2F%2Fwww.ourcommons.ca text/html; charset=utf-8 327158
399036 [ZAP-PassiveScan-2] WARN  org.zaproxy.zap.extension.pscan.PassiveScanTask - Passive Scan rule Vulnerable JS Library (Retire.js) took 11 seconds to scan https://platform.twitter.com/widgets/widget_iframe.d7fc2fc075c61f6fa34d79a0cbbf1e34.html?origin=https%3A%2F%2Fwww.ourcommons.ca text/html; charset=utf-8 327158
```

Reply:
Great, no need to add a unit test IMO.

Reply:
Okay, I'll look into it. So far 20MB of lorem ipsum doesn't slow it down (this or the original), might actually need to be HTML.

Reply:
> Great, no need to add a unit test IMO.

Thanks